Source/ASCollectionView.mm

```diff
@@ -1930,6 +1972,11 @@ - (void)layer:(CALayer *)layer didChangeBoundsWithOldValue:(CGRect)oldBounds new
   if (self.collectionViewLayout == nil) {
     return;
   }
+  if (_collectionViewLayoutFlags.providesLayoutCalculator) {
+    // TODO Maybe kick off a background calculation here and hope that the layout is ready by the time -prepareLayout
```

Comment:
One risk of this is multiple bounds changes.  We should probably schedule it to occur at the end of the runloop if anything, but that will probably mean that layout is called by then anyway.  Because the main thread runs at a higher priority, I'm not sure we would gain too much from starting it early -- although there could be other work on the main thread to run before layout is flushed.

The other question is whether there are sometimes offscreen collections that have their bounds set up, but that would be wasteful to actually run the layout for.  We could check for that here, but something to consider.

For the sake of simplicity and ensuring reliability before we introduce new layout kick-off points, it probably makes sense to defer this until testing indicates there are some reasonably-common use cases where this would provide an advantage.